exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/InnerTableLinkEntity.kt

```diff
@@ -0,0 +1,54 @@
+package org.jetbrains.exposed.dao
+
+import org.jetbrains.exposed.dao.id.EntityID
+import org.jetbrains.exposed.dao.id.IdTable
+import org.jetbrains.exposed.sql.Column
+import org.jetbrains.exposed.sql.ResultRow
+
+/**
+ * Base class for an [Entity] instance identified by a [wrapped] entity comprised of any ID value.
+ *
+ * Instances of this base class should be used when needing to represent referenced entities in a many-to-many relation
+ * from fields defined using `via`, which require additional columns in the intermediate table. These additional
+ * columns should be added as constructor properties and the property-column mapping should be defined by
+ * [getInnerTableLinkValue].
+ *
+ * @param WID ID type of the [wrapped] entity instance.
+ * @property wrapped The referenced (parent) entity whose unique ID value identifies this [InnerTableLinkEntity] instance.
+ * @sample org.jetbrains.exposed.sql.tests.shared.entities.ViaTests.ProjectWithData
+ */
+abstract class InnerTableLinkEntity<WID : Comparable<WID>>(val wrapped: Entity<WID>) : Entity<WID>(wrapped.id) {
+    /**
+     * Returns the initial column-property mapping for an [InnerTableLinkEntity] instance
+     * before being flushed and inserted into the database.
+     *
+     * @sample org.jetbrains.exposed.sql.tests.shared.entities.ViaTests.ProjectWithData
+     */
+    abstract fun getInnerTableLinkValue(column: Column<*>): Any?
```

Comment:
Even though an entity class is being used to accomplish this feature, the original behavior and usage of `via()` entities should most likely be the same. I think it would be best to override all standard entity functions, like `delete()`, `all()`, `new()`, `findById()`, etc since their use wouldn't work (since the intermediate table does not have to be an `IdTable`).

This would mean that attempting to call them on an `InnerTableLinkEntity` throws an error instead. So the only way to insert or delete or retrieve these linked values would be by setting/getting the parent/child entity field declared with `via()`. Basically these entities would only exist or be used through the `via` field. Does that make sense?